tokio/src/util/linked_list.rs

```diff
@@ -313,6 +347,146 @@ cfg_taskdump! {
     }
 }
 
+// ===== impl AtomicLinkedList =====
+
+feature! {
+    #![feature = "sync"]
+
+    use core::sync::atomic::{
+        AtomicPtr,
+        Ordering::{AcqRel, Relaxed},
+    };
+
```

Comment:
Please split this into a separate file. Rustfmt doesn't work inside the macros, so I don't want large codeblocks inside them.

Reply:
I created a separate file in the least intrusive way I could think of. A better approach I think would be to refactor linked_list.rs into multiple files, but probably it is better done outside of this PR. Please let me know if you have a better idea of how it could be done.